streams/src/main/java/org/apache/kafka/streams/state/internals/StreamsMetadataImpl.java

```diff
@@ -0,0 +1,163 @@
+                                                                            Collections.emptySet(),
+                                                                            Collections.emptySet(),
+                                                                            Collections.emptySet());
+
+    private final HostInfo hostInfo;
+
+    private final Set<String> stateStoreNames;
+
+    private final Set<TopicPartition> topicPartitions;
+
+    private final Set<String> standbyStateStoreNames;
+
+    private final Set<TopicPartition> standbyTopicPartitions;
+
+    public StreamsMetadataImpl(final HostInfo hostInfo,
+                               final Set<String> stateStoreNames,
+                               final Set<TopicPartition> topicPartitions,
+                               final Set<String> standbyStateStoreNames,
+                               final Set<TopicPartition> standbyTopicPartitions) {
+
+        this.hostInfo = hostInfo;
+        this.stateStoreNames = stateStoreNames;
+        this.topicPartitions = topicPartitions;
+        this.standbyTopicPartitions = standbyTopicPartitions;
+        this.standbyStateStoreNames = standbyStateStoreNames;
+    }
+
+    /**
+     * The value of {@link org.apache.kafka.streams.StreamsConfig#APPLICATION_SERVER_CONFIG} configured for the streams
+     * instance, which is typically host/port
+     *
+     * @return {@link HostInfo} corresponding to the streams instance
+     */
+    @Override
+    public HostInfo hostInfo() {
+        return hostInfo;
+    }
+
+    /**
+     * State stores owned by the instance as an active replica
+     *
+     * @return set of active state store names
+     */
+    @Override
+    public Set<String> stateStoreNames() {
+        return Collections.unmodifiableSet(stateStoreNames);
+    }
+
+    /**
+     * Topic partitions consumed by the instance as an active replica
+     *
+     * @return set of active topic partitions
+     */
+    @Override
+    public Set<TopicPartition> topicPartitions() {
+        return Collections.unmodifiableSet(topicPartitions);
+    }
+
+    /**
+     * (Source) Topic partitions for which the instance acts as standby.
+     *
+     * @return set of standby topic partitions
+     */
+    @Override
+    public Set<TopicPartition> standbyTopicPartitions() {
+        return Collections.unmodifiableSet(standbyTopicPartitions);
+    }
+
+    /**
+     * State stores owned by the instance as a standby replica
+     *
+     * @return set of standby state store names
+     */
+    @Override
+    public Set<String> standbyStateStoreNames() {
+        return Collections.unmodifiableSet(standbyStateStoreNames);
+    }
+
+    @Override
+    public String host() {
+        return hostInfo.host();
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public int port() {
+        return hostInfo.port();
+    }
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Could you add unit tests for equals() and hashCode()? Since we guarantee a behavior on the interface, it makes sense to verify if the behavior of the implementation is correct.